core/tests/test-dependencies/root-files-mapping-generator.ts

```diff
@@ -0,0 +1,723 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Script to generate a files to root files mapping for the
+ * Oppia codebase.
+ */
+
+import path from 'path';
+import fs from 'fs';
+import {Decorator, SourceFile, ts} from 'ts-morph';
+import * as cheerio from 'cheerio';
+import {
+  project,
+  ROOT_DIRECTORY,
+  AngularDecorators,
+  getRelativePathToRootDirectory,
+  isNodeModule,
+  getDecorationNodesByTextFromSourceFile,
+  resolveModuleRelativeToRoot,
+  getDecorationNodeText,
+  getValueFromLiteralStringOrBinaryExpression,
+} from './typescript-ast-utilities';
+import {
+  getPageModules,
+  getPageModulesCoveredByRouteMapping,
+} from './route-to-module-mapping-generator';
+
+interface BaseAngularInformation {
+  className: string;
+}
+
+interface AngularModuleInformation extends BaseAngularInformation {
+  type: 'module';
+}
+
+interface AngularComponentInformation extends BaseAngularInformation {
+  type: 'component';
+  selector?: string;
+  templateFilePath?: string;
+}
+
+interface AngularDirectiveOrPipeInformation extends BaseAngularInformation {
+  type: 'directive' | 'pipe';
+  selector?: string;
+}
+
+type AngularInformation =
+  | AngularModuleInformation
+  | AngularComponentInformation
+  | AngularDirectiveOrPipeInformation;
+
+const GIT_IGNORED_EXCLUSIONS = fs
+  .readFileSync(path.resolve(ROOT_DIRECTORY, '.gitignore'), 'utf-8')
+  .split('\n')
+  .filter(line => line.trim() && !line.startsWith('#'));
+
+const FILE_EXCLUSIONS = [
```

Comment:
What if I as a developer add a new file to the codebase that should be excluded, but I do not know it should be excluded, what happens then?

Reply:
It would be rare to exclude something in file exclusions and I don't really see any concrete way to guide a developer to excluding something.